dgm-builder/src/main/java/com/cloudbees/groovy/cps/tool/Translator.java

```diff
@@ -333,20 +336,40 @@ public JExpression visitMethodInvocation(MethodInvocationTree mt, Void __) {
 
                 if (ms instanceof MemberSelectTree) {
                     MemberSelectTree mst = (MemberSelectTree) ms;
-                    inv = $b.invoke("functionCall")
-                        .arg(loc(mt))
-                        .arg(visit(mst.getExpression()))
-                        .arg(n(mst.getIdentifier()));
+                    // Make sure we translate any method calls to a transformed class too.
+                    if (mst.getExpression() instanceof JCIdent &&
+                            !((JCIdent)mst.getExpression()).sym.toString().equals(fqcn) &&
+                            otherTranslated.containsKey(((JCIdent)mst.getExpression()).sym.toString())) {
+                        inv = $b.invoke("functionCall")
+                                .arg(loc(mt))
+                                .arg($b.invoke("constant").arg(
+                                        otherTranslated.get(((JCIdent)mst.getExpression()).sym.toString()).dotclass()))
+                                .arg(n(mst.getIdentifier()));
+
+                    } else {
```

Comment:
So I have _no clue_ what this change is about—can you add some comments to the PR diff giving background?